workshops/util.py

```diff
@@ -167,18 +170,60 @@ def verify_upload_person_task(data):
                         Task.objects.get(event=existing_event, person=person,
                                          role=existing_role)
                     except Task.DoesNotExist:
-                        info.append('Task will be created')
+                        info.append('Task will be created.')
                     else:
-                        info.append('Task already exists')
+                        info.append('Task already exists.')
+        else:
+            info.append('It\'s highly recommended to add an email address.')
 
         if person:
-            if not any([event, role]):
-                errors.append("User exists but no event and role to assign to"
-                              " the user to was provided")
+            # force username from existing record
+            item['username'] = person.username
+            item['person_exists'] = True
 
-        if (event and not role) or (role and not event):
-            errors.append("Must have both: event ({0}) and role ({1})"
-                          .format(event, role))
+        else:
+            # force a newly created username
+            if not item.get('username'):
+                item['username'] = create_username(personal, family)
```

Comment:
`verify_upload_person_task` is called multiple times, for example upon clicking 'Verify' button. See `person_bulk_add_confirmation` view for details.